### PR TITLE
Fix dark mode detection when using auto-switching for dark mode

### PR DESCRIPTION
**Pull request by anderscarling**

### CODE REVIEW

File: Sources/DarkMode.swift [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+import AppKit
 
 struct DarkMode {
 	private static let prefix = "tell application \"System Events\" to tell appearance preferences to"
 
 	static var isEnabled: Bool {
 		get {
-			UserDefaults.standard.string(forKey: "AppleInterfaceStyle") == "Dark"
+			if #available(macOS 10.14, *) {
+				return NSAppearance.current.name == .darkAqua
```
Comment (sindresorhus) [inline Sources/DarkMode.swift:10]: This is not correct. It doesn't take into accessibility settings.

```suggestion
				return NSAppearance.current.bestMatch(from: [.darkAqua, .aqua]) == .darkAqua
```

Comment (anderscarling) [reply Sources/DarkMode.swift:10]: Great catch, reading the docs I managed to scan over the accessibility lines thinking they where also related to visual effects (as vibrantLight and vibrantDark is). Will incorporate your suggested change.

**Comment by anderscarling** [comment]
Weirdly, my old code worked just as well as using the `bestMatch` api to check for appearance, even though I enabled "increased contrast" and maxed out the contrast slider in accessibility settings.

Screenshot 2021-03-05 at 11 21 54 AM
 
Tried a bit of googling but could not find any information about how to trigger .accessibilityHighContrastDarkAqua so I was not able to find a configuration where the old code broke.

However, I agree using bestMatch is definitely the way to go and seems to work in all scenarios I was able to reproduce.

**Comment by sindresorhus** [comment]
> Tried a bit of googling but could not find any information about how to trigger .accessibilityHighContrastDarkAqua so I was not able to find a configuration where the old code broke.

I don't know either, but Apple explicitly said in a WWDC video at the time that `bestMatch` was the correct way to detect.

**Pull request merged**